forgerock-auth/src/androidTest/java/org/forgerock/android/auth/callback/DeviceSigningVerifierApplicationPinCallbackTest.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.callback;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import androidx.test.core.app.ActivityScenario;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+
+import org.assertj.core.api.Assertions;
+import org.forgerock.android.auth.DummyActivity;
+import org.forgerock.android.auth.FRListener;
+import org.forgerock.android.auth.FRSession;
+import org.forgerock.android.auth.InitProvider;
+import org.forgerock.android.auth.Logger;
+import org.forgerock.android.auth.Node;
+import org.forgerock.android.auth.NodeListener;
+import org.forgerock.android.auth.NodeListenerFuture;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.concurrent.ExecutionException;
+
+public class DeviceSigningVerifierApplicationPinCallbackTest extends BaseDeviceBindingTest {
+    protected final static String TREE = "device-verifier";
+    protected final static String APPLICATION_PIN = "1234";
+
+    @BeforeClass
+    public static void bindDevice() throws ExecutionException, InterruptedException {
+        final int[] bindSuccess = {0};
+        CallbackFactory.getInstance().register(CustomApplicationPinDeviceBindingCallback.class);
+
+        ActivityScenario scenario = ActivityScenario.launch(DummyActivity.class);
+        scenario.onActivity(InitProvider::setCurrentActivity);
+
+        NodeListenerFuture<FRSession> nodeListenerFuture = new DeviceSigningVerifierNodeListener(context, "bind-pin")
+        {
+            NodeListener<FRSession> nodeListener = this;
```

Comment:
This can be final, you may want to change other tests which have similar pattern.